src/setup.ts

```diff
@@ -45,12 +45,10 @@ export const setup = async (argv: Config) => {
 
   const header = await api.getHeader(blockHash)
 
-  const blockNumber = +header.number
-  const timestamp = argv.timestamp ?? Date.now()
-  const setTimestamp = new SetTimestamp((newBlockNumber) => {
-    return timestamp + (newBlockNumber - blockNumber) * 12000 // TODO: make this more flexible
-  })
+  // TODO: do we really need to set a custom timestamp?
+  // const timestamp = argv.timestamp
```

Comment:
yes we do. some of Acala honzon logic depends on timestamp and we want to be able to mock it

Reply:
in that case we need to have a helper method that modify slot and timestamp storage

Reply:
SetTimestamp doesn't have to know anything. It will just produce valid timestamps

Reply:
I guess we could have a SetExactTimestamp or something and allow config to choose what inherents to include

Reply:
it won't work will latest upgrade,  that's why I updated this